compile/compile.go

```diff
@@ -430,6 +427,45 @@ func (c *Compiler) compileWasm(ctx context.Context) error {
 
 	c.bundle.Wasm = buf.Bytes()
 
+	modulePath := bundle.WasmFile
+
+	c.bundle.WasmModules = []bundle.WasmModuleFile{{
+		URL:  modulePath,
+		Path: modulePath,
+		Raw:  buf.Bytes(),
+	}}
+
+	// Each entrypoint needs an entry in the manifest along with the
+	// original rule(s) removed from the remaining rego modules.
+	for i, entrypoint := range c.entrypointrefs {
+		entrypointPath := c.entrypoints[i]
+
+		c.bundle.Manifest.WasmResolvers = append(c.bundle.Manifest.WasmResolvers, bundle.WasmResolver{
+			Module:     "/" + strings.TrimLeft(modulePath, "/"),
+			Entrypoint: entrypointPath,
+		})
+
+		for i := 0; i < len(c.bundle.Modules); i++ {
+			mf := &c.bundle.Modules[i]
+
+			// Drop any rules that match the entrypoint path.
+			var rules []*ast.Rule
+			for _, rule := range mf.Parsed.Rules {
+				if !rule.Path().Equal(entrypoint.Value) {
+					rules = append(rules, rule)
+				}
+			}
+
+			// If any rules were dropped update the module accordingly
+			if len(rules) != len(mf.Parsed.Rules) {
+				mf.Parsed.Rules = rules
+				// Remove the original raw source, we're editing the AST
+				// directly so it wont be in sync anymore.
+				mf.Raw = nil
+			}
+		}
```

Comment:
These changes will leave the bundle in an invalid state if there are unresolved references to the entrypoint. For example:

example.rego:

```rego
package foo

p { q > 7 }
q = 6
```

If the entrypoint was `foo/q` then the resulting example.rego file in the output bundle would be:

```rego
package foo
p { q > 7}
```

This version of `p` is unsafe because `q` is not defined anywhere. We could either resolve the refs when we remove the rules or we could modify the ast.Compiler to resolve refs based on externs. I prefer the former because otherwise we're allowing for .rego files emitted by the compiler to be invalid on their own.

Reply:
Updated in latest changes to inject the import statements